test/integration/resume_request_test.go

```diff
@@ -0,0 +1,183 @@
+	// can be returned instead of a deadline exceeded error.
+	if err == nil {
+		t.Errorf("expected error but got nil")
+	}
+
+	// ensure a single CertificateRequest exists, and fetch its UID so we can compare it later
+	existingRequestUID := ensureOneRequestExists(ctx, t, opts.Client, ns, "")
+
+	// run NodePublishVolume once again, with a short timeout.
+	// here we want to ensure that no second request is completed, and the timeout is reached again.
+	// we still won't actually complete issuance here.
+	twoSecondCtx, cancel = context.WithTimeout(ctx, time.Second*2)
+	defer cancel()
+	_, err = cl.NodePublishVolume(twoSecondCtx, nodePublishVolumeRequest)
+	// assert that an error has been returned - we don't mind what kind of error, as due to the async nature of
+	// de-registering metadata from the metadata store upon failures, there is a slim chance that a metadata read error
+	// can be returned instead of a deadline exceeded error.
+	if err == nil {
+		t.Errorf("expected error but got nil")
+	}
+	// ensure the same certificaterequest object still exists
+	ensureOneRequestExists(ctx, t, opts.Client, ns, existingRequestUID)
+
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	if issueBeforeCall {
+		// we don't run this in a goroutine so we can be sure the certificaterequest is completed BEFORE the issue loop is entered
+		testutil.IssueOneRequest(t, opts.Client, "certificaterequest-namespace", stopCh, selfSignedExampleCertificate, []byte("ca bytes"))
+	} else {
+		go func() {
+			// allow 500ms before actually issuing the request so we can be sure we're within the issue() function call
+			// when the certificaterequest is finally completed
+			time.Sleep(time.Millisecond * 500)
+			testutil.IssueOneRequest(t, opts.Client, "certificaterequest-namespace", stopCh, selfSignedExampleCertificate, []byte("ca bytes"))
+		}()
+	}
+
+	// call NodePublishVolume again. this time, we expect NodePublishVolume to return without an error and actually issue
+	// the certificate using the existing request data.
+	// We don't use an explicit timeout here to avoid any weird race conditions caused by shorter test timeouts.
+	_, err = cl.NodePublishVolume(ctx, nodePublishVolumeRequest)
+	if err != nil {
+		t.Errorf("expected no error but got: %v", err)
+	}
+	// ensure the same certificaterequest object still exists
+	ensureOneRequestExists(ctx, t, opts.Client, ns, existingRequestUID)
+
+	files, err := store.ReadFiles("test-vol")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !reflect.DeepEqual(files["ca"], []byte("ca bytes")) {
+		t.Errorf("unexpected CA data: %v", files["ca"])
+	}
+	if !reflect.DeepEqual(files["cert"], selfSignedExampleCertificate) {
+		t.Errorf("unexpected certificate data: %v", files["cert"])
+	}
+}
+
+func TestResumesExistingRequest_IssuedBetweenPublishCalls(t *testing.T) {
+	testResumesExistingRequest(t, true)
+}
+
+func TestResumesExistingRequest_IssuedDuringPublishCall(t *testing.T) {
+	testResumesExistingRequest(t, false)
+}
```

Comment:
Consider using a type bool enum or passing an options struct so that it is clearer to the reader what `true` and `false` is doing here.